File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    static String alph[] = {"A", "E", "I", "O", "U"};
    static ArrayList<String> list;
     
    public int solution(String word) {
        int answer = 0;
        list = new ArrayList<>();
        dfs("", 0);
        
        for(int i=0; i<list.size(); i++) {
            if(list.get(i).equals(word)) {
                answer = i;
            }
        }
        
        return answer;
    }
    
    public void dfs(String str, int len) {
        list.add(str);
        if(len == 5) return; //5이상의 len 붙는건 막기 
        for(int i=0; i< 5; i++) {
           dfs(str+alph[i], len+1);
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;
int answer = 0;
string arr = "AEIOU";
string w;
int cnt = 0;

void dfs(int sIdx, string s) {
    if(answer != 0 || sIdx == 6) return;
    if(s != "") cnt++;
    
    if(s == w) {
        answer = cnt;
        return;
    }
    
    for(int i=0; i<5; i++) {
        dfs(sIdx+1 ,s+arr[i]);
    }
}

int solution(string word) {
    w = word;
    dfs(0,"");
    return answer;
}